src/test_heartbeat_led.sh

```diff
@@ -0,0 +1,152 @@
+#!/bin/sh
+#
+# Copyright (c) 2016, Imagination Technologies Limited and/or its affiliated group companies
+# and/or licensors
+#
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without modification, are permitted
+# provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this list of conditions
+#    and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice, this list of
+#    conditions and the following disclaimer in the documentation and/or other materials provided
+#    with the distribution.
+#
+# 3. Neither the name of the copyright holder nor the names of its contributors may be used to
+#    endorse or promote products derived from this software without specific prior written
+#    permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+# IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+# FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+# WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+# This test will blink the Heartbeat LED on the board
+
+. /usr/lib/board_test_common.sh
+
+BLINK_TIME=10
+PASS=0
+FAIL=0
+
+USAGE="
+Usage: $0 options
+
+OPTIONS:
+-h         Show this message
+-t  <arg>  Number of seconds to blink heartbeat led, default 10, and pass -t 0 for continuous mode
+-v         Verbose
+-V         Show package version
+"
+
+while getopts "t:vVh" opt; do
+	case $opt in
+		t)
+			BLINK_TIME=$OPTARG
+			;;
+		v)
+			LOG_LEVEL=2
+			;;
+		V)
+			echo "version = "$(cat $BOARD_TEST_PATH/version)
+			exit $SUCCESS
+			;;
+		h)
+			echo "${USAGE}"
+			exit $SUCCESS
+			;;
+		\?)
+			echo "${USAGE}"
+			exit $FAILURE
+			;;
+	esac
+done
+
+redirect_output_and_error $LOG_LEVEL
+
+disable_heartbeat_led()
+{
+	echo "none" > /sys/class/leds/marduk:red:heartbeat/trigger
```

Comment:
should we preserve the previous status while exiting ?
By default heartbeat/trigger is "timer". while coming out of this heartbeat test, we can set the "timer" again, instead of "none".